src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -192,10 +192,43 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 			this._changeRef.detectChanges();
 		}));
 		this._register(this.cellModel.onCellPreviewModeChanged(preview => {
+			// On preview mode change, get the cursor position (get the position only when the selection node is a text node)
+			if (!preview && window.getSelection() && window.getSelection().focusNode?.nodeName === '#text' && window.getSelection().getRangeAt(0)) {
+				let selection = window.getSelection().getRangeAt(0);
+
+				let startElementNodes = [];
+				let startNode = selection.startContainer;
+				let endNode = selection.endContainer;
+				while (startNode !== this.output.nativeElement) {
+					startElementNodes.push(this.getNodeIndex(startNode));
+					startNode = startNode.parentNode;
+				}
+				let endElementNodes = [];
+				while (endNode !== this.output.nativeElement) {
+					endElementNodes.push(this.getNodeIndex(endNode));
+					endNode = endNode.parentNode;
+				}
+				// Create cursor position
+				let cursorPosition: ICaretPosition = {
+					startElementNodes: startElementNodes,
+					startOffset: selection.startOffset,
+					endElementNodes: endElementNodes,
+					endOffset: selection.endOffset
+				};
+				this.cellModel.richTextCursorPosition = cursorPosition;
+			}
 			this.previewMode = preview;
 			this.focusIfPreviewMode();
 		}));
 		this._register(this.cellModel.onCellMarkdownModeChanged(markdown => {
+			if (!markdown && window.getSelection() && window.getSelection().getRangeAt(0)) {
+				let editorControl = this.cellEditors.length > 0 ? this.cellEditors[0].getEditor().getControl() : undefined;
+				if (editorControl) {
+					let selection = editorControl.getSelections()[0];
+
```

Comment:
I think you can use getSelection() instead, since we are getting the first one anyways